spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,122 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+
+  describe "Logged in user" do
+    login_user
+    before :each do
+      @shop = create(:shop, user: subject.current_user)
+      p @shop
+    end
+
+    it "goes to all product page" do
+      get :index, shop_id: @shop.id
+      expect(response).to be_success
+      expect(response).to have_http_status(200)
+    end
+
+    it "goes to the creating product page" do
+      get :new, shop_id: @shop.id
+      expect(response).to be_success
+      expect(response).to have_http_status(200)
+    end
+
+    it "creates a product" do
+      product = attributes_for(:product, shop: @shop)
+      p product
+      post :create, { shop_id: @shop.id, product: product }
+      expect(response).to be_success
+      expect(response).to have_http_status(200)
+    end
+
+    it "shows the specified product" do
+      product = create(:product, shop: @shop)
+      p product
+      get :show, { shop_id: @shop.id, id: product.id }
+      expect(response).to be_success
+      expect(response).to have_http_status(200)
+    end
+
+    it "goes to the editing product page" do
+      product = create(:product, shop: @shop)
```

Comment:
Could use `contexts` to group these better and run common functions in a before function?